rust/xaynet-sdk/src/state_machine/state_machine.rs

```diff
@@ -65,6 +66,17 @@ impl StateMachine {
             StateMachine::Sum2(phase) => phase.state.into(),
         }
     }
+
+    /// Return the model configuration of the model that is expected in the update phase.
+    pub fn model_config(&self) -> ModelConfig {
```

Comment:
@little-dude @finiteprods 
What do you think about the idea that we pass the local model config to the user via the notifier (maybe in `IO::notify_load_model`) instead of a state machine method?    

The problem I see here with the state machine method is that the state machine can no longer simply be executed in a loop like:

```rust
        loop {
            self = match self.0.transition().await {
                TransitionOutcome::Pending(state_machine) => {
                    delay_for(tick).await;
                    Self(state_machine)
                }
                TransitionOutcome::Complete(state_machine) => Self(state_machine),
            };
        }
```

If the user wants to receive the model configuration, he has to interrupt the loop in order to call the method

But I would like to do this once we included the data type to the global model so that I don't depend on the local model for converting the global model